agent/app/agent_capability_test.go

```diff
@@ -142,6 +128,93 @@ func TestCapabilities(t *testing.T) {
 	}
 }
 
+// Test on-prem capability by checking that when on-prem config is set, unsupported capabilities aren't
+// added, and on-prem capability is added.
+func TestCapabilitiesOnPrem(t *testing.T) {
```

Comment:
It's good that we test that the unsupported capabilities are not present, but should we also test that the expected supported capabilities are there?
If in the future we add a new capability that should be on prem but for some error in logic it's ommited, nothing would tell us there's something wrong